website/src/components/FlaggableElement.tsx

```diff
@@ -0,0 +1,205 @@
+};
+type FlagRefs = {
+  label: string;
+  value: number;
+  setValue;
+  isChecked: boolean;
+  setChecked;
+};
+
+export const FlaggableElement = (props) => {
+  const [isEditing, setIsEditing] = useBoolean();
+  const [hasCheck, setHasCheck] = useBoolean();
+  const { trigger, isMutating } = useSWRMutation("/api/v1/text_labels", poster, {
+    onSuccess: () => {
+      setIsEditing.off;
+    },
+  });
+
+  const submitResponse = (label_data) => {
+    trigger(label_data);
+  };
+  const label_refs: FlagRefs[] = TEXT_LABEL_FLAGS.map((flag) => {
+    const [isChecked, setChecked] = useState(false);
+    const [value, setValue] = useState(1);
+    return {
+      label: flag.attributeName,
+      value: value,
+      setValue: setValue,
+      isChecked: isChecked,
+      setChecked: setChecked,
+    };
+  });
+  const fetchData = () => {
+    let label_map: Map<string, number> = new Map();
+    label_refs
+      .filter((flagRef) => flagRef.isChecked)
+      .map((refs) => {
+        label_map.set(refs.label, refs.value);
+      });
+    console.log(label_map);
+    return { post_id: props.post_id, label_map: Object.fromEntries(label_map), text: props.text };
+  };
+
+  return (
+    <Popover
+      isOpen={isEditing}
+      onOpen={setIsEditing.on}
+      onClose={setIsEditing.off}
+      closeOnBlur={false}
+      isLazy
+      lazyBehavior="keepMounted"
+    >
+      <div className="inline-block float-left">
+        <PopoverAnchor>{props.children}</PopoverAnchor>
+        <PopoverTrigger>
+          <Button h="20px">
+            <FlagIcon className="h-5 w-5 ml-3 text-gray-400 group-hover:text-gray-500" aria-hidden="true" />
+          </Button>
+        </PopoverTrigger>
+      </div>
+
+      <PopoverContent width="fit-content">
+        <PopoverArrow />
+        <PopoverCloseButton />
+        <div className="flex mt-3 ">
+          <PopoverBody>
+            <ul>
+              {TEXT_LABEL_FLAGS.map((option, i) => {
+                return <FlagCheckboxLi option={option} key={i} state={label_refs[i]}></FlagCheckboxLi>;
+              })}
+            </ul>
+            <div className="flex justify-center ml-auto">
+              <Button
+                onClick={() => submitResponse(fetchData())}
+                className="bg-indigo-600 text-black hover:bg-indigo-700"
+              >
+                Report
+              </Button>
+            </div>
+          </PopoverBody>
+        </div>
+      </PopoverContent>
+    </Popover>
+  );
+};
+function FlagCheckboxLi(props: { option: textFlagLabels; state: FlagRefs }): JSX.Element {
+  let AdditionalExplanation = null;
+  if (props.option.additionalExplanation) {
+    AdditionalExplanation = (
+      <a href="#" className="group flex items-center space-x-2.5 text-sm ">
+        <QuestionMarkCircleIcon
+          className="flex h-5 w-5 ml-3 text-gray-400 group-hover:text-gray-500"
+          aria-hidden="true"
+        />
+      </a>
+    );
+  }
+
+  return (
+    <>
```

Comment:
I think we can drop the surrounding `<> ...</>` since this is all in a single `<li>`

Reply:
thank you! have removed